diff --git a/lib/eeprom/ethtool-util.h b/lib/eeprom/ethtool-util.h
index 3fe47e3..8e6f463 100644
--- a/lib/eeprom/ethtool-util.h
+++ b/lib/eeprom/ethtool-util.h
@@ -22,19 +22,19 @@
 #ifndef ETHTOOL_UTIL_H__
 #define ETHTOOL_UTIL_H__
 
-#include <sys/types.h>
+#include <stdint.h>
 
 /* hack, so we may include kernel's ethtool.h */
 typedef unsigned long long __u64;
-typedef __uint32_t __u32;         /* ditto */
-typedef __uint16_t __u16;         /* ditto */
-typedef __uint8_t __u8;           /* ditto */
+typedef uint32_t __u32;         /* ditto */
+typedef uint16_t __u16;         /* ditto */
+typedef uint8_t __u8;           /* ditto */
 
 /* historical: we used to use kernel-like types; remove these once cleaned */
 typedef unsigned long long u64;
-typedef __uint32_t u32;         /* ditto */
-typedef __uint16_t u16;         /* ditto */
-typedef __uint8_t u8;           /* ditto */
+typedef uint32_t u32;         /* ditto */
+typedef uint16_t u16;         /* ditto */
+typedef uint8_t u8;           /* ditto */
 
 #include <net/if.h>
 
